Classes/Controller/Haiku/DetailController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace T3docs\Examples\Controller\Haiku;
+
+use Psr\Http\Message\ServerRequestInterface;
+use T3docs\Examples\Domain\Repository\HaikuRepository;
+use T3docs\Examples\Service\FlexFormSettingsService;
+use T3docs\Examples\Service\StandaloneViewService;
+use TYPO3\CMS\Core\Http\PropagateResponseException;
+use TYPO3\CMS\Core\Http\Response;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+
+class DetailController
+{
+    /**
+     * The back-reference to the mother cObj object set at call time
+     */
+    public ContentObjectRenderer $cObj;
+    public array $conf = [];
+    public StandaloneView $view;
+
+    public function __construct(
+        private readonly HaikuRepository $haikuRepository,
+        private readonly FlexFormSettingsService $flexFormSettingsService,
+        private readonly StandaloneViewService $viewService,
+    ) {
+    }
+
+    /**
+     * This setter is called when the plugin is called from UserContentObject (USER)
+     * via ContentObjectRenderer->callUserFunction().
+     */
+    public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
+    {
+        $this->cObj = $cObj;
+    }
+
+    public function main(string $content, array $conf, ServerRequestInterface $request): string
+    {
+        $this->conf = $conf;
+        $this->loadFlexFormSettings();
+        $this->view = $this->viewService->createView($this->conf, 'Haiku/List');
+
+        $parameter = $request->getQueryParams()['tx_examples_haiku']??[];
+        $action = $parameter['action'] ?? 'list';
+        return match ($action) {
+            'show' => $this->showAction((int)$parameter['haiku'] ?? 0),
```

Comment:
```suggestion
            'show' => $this->showAction((int)($parameter['haiku'] ?? 0)),
```

Casting has precedence over null coalescing operator - that is not what we want, therefore the brackets.